src/vm/runners/cairo_runner.rs

```diff
@@ -94,21 +105,70 @@ impl CairoRunner {
+                    builtin_name.clone(),
+                    HashBuiltinRunner::new(
+                        self.layout
+                            .builtins
+                            .pedersen
+                            .as_ref()
+                            .ok_or_else(|| {
+                                RunnerError::NoBuiltinForInstance(
+                                    self.layout.name.clone(),
+                                    builtin_name.to_string(),
+                                )
+                            })?
+                            .ratio
+                            .to_owned(),
+                    )
+                    .into(),
+                ));
             }
 
             if builtin_name == "range_check" {
-                //Information for Buitin info taken from here https://github.com/starkware-libs/cairo-lang/blob/b614d1867c64f3fb2cf4a4879348cfcf87c3a5a7/src/starkware/cairo/lang/instances.py#L115
+                let range_check_instance =
+                    self.layout.builtins.range_check.as_ref().ok_or_else(|| {
+                        RunnerError::NoBuiltinForInstance(
+                            self.layout.name.clone(),
+                            builtin_name.to_string(),
+                        )
+                    })?;
                 builtin_runners.push((
                     builtin_name.clone(),
-                    RangeCheckBuiltinRunner::new(bigint!(8), 8).into(),
+                    RangeCheckBuiltinRunner::new(
+                        range_check_instance.ratio,
+                        range_check_instance.n_parts,
+                    )
+                    .into(),
                 ));
             }
             if builtin_name == "bitwise" {
-                builtin_runners.push((builtin_name.clone(), BitwiseBuiltinRunner::new(256).into()));
+                builtin_runners.push((
+                    builtin_name.clone(),
+                    BitwiseBuiltinRunner::new(self.layout.builtins.bitwise.as_ref().ok_or_else(
+                        || {
+                            RunnerError::NoBuiltinForInstance(
+                                self.layout.name.clone(),
+                                builtin_name.to_string(),
+                            )
+                        },
+                    )?)
+                    .into(),
+                ));
             }
             if builtin_name == "ec_op" {
-                builtin_runners.push((builtin_name.clone(), EcOpBuiltinRunner::new(256).into()));
+                builtin_runners.push((
+                    builtin_name.clone(),
+                    EcOpBuiltinRunner::new(self.layout.builtins.ec_op.as_ref().ok_or_else(
+                        || {
+                            RunnerError::NoBuiltinForInstance(
+                                self.layout.name.clone(),
+                                builtin_name.to_string(),
+                            )
+                        },
```

Comment:
This lambda is repeated many times, it may be a good idea to just extract it as a variable.

Reply:
I don't understand this comment, We have to do this check for every builtin which uses the corresponding builtin instance data in its creation, we can't just do it once

Reply:
The check, yes. The lambda, no. AFAIK lambdas can be assigned, so you can do:
```
let nobuiltin = || {
    RunnerError::NoBuiltinForInstance(
        self.layout.name.clone(),
        builtin_name.to_string(),
    )
};
```
Then you just pass `nobuiltin` in the call places.

Reply:
I understand now, thanks for explaining! good call!

Reply:
Changed(6990cf9)